### PR TITLE
rework outbits and DIR

### PR DESCRIPTION
**Pull request by tonnico**
We get a free Dir-High delay. Could be good for some slow stepper driver. Maybe this could also help THB6128. #1041

### CODE REVIEW

**Comment by thinkyhead** [comment]
The diff is hard to follow – I'll have to look at this in some other compare tool to uncross my eyes....

**Comment by Wurstnase** [comment]
I will make an update soon.

**Comment by tonnico** [comment]
@thinkyhead The diff is relativ simple. I took every direction-setup and put them in a new function. We don't need to check the direction while homing. The homing itself sets the direction because it's a move like every move. Also the direction will never change in a trapez, thats bresenham.

The rest in the stepper isr is the check endstop routine.

Also i made a check out_bits routine. Only if the out_bits changed, the direction are set. Especially for curves this could improve the speed a tiny bit.

Maybe G2/G3 needs them?!? Don't test them.

**Comment by thinkyhead** [comment]
@Wurstnase G2 and G3 are probably fine. The arc functions just call the planner's line function in a loop.

**Comment by tonnico** [comment]
Ok.  That's good. But wait with the merge. I need some time for testing. Got some strange bug with BAL.

**Comment by thinkyhead** [comment]
At least I see all your commits are passing Travis. I have to merge commits left and right to cover my sloppiness. My favorite new way to make a clean branch for a PR, skipping all that Merge and Rebase nonsense:

```
git checkout Development -b my_feature_clean
git diff Development origin/my_feature_messy | git apply
```

**Comment by tonnico** [comment]
Will make it tomorrow.

**Comment by thinkyhead** [comment]
@Wurstnase So everything checks out ok with this now? I'm looking forward to integrating it!

**Comment by tonnico** [comment]
I give you an OK later. I was on holiday and don't test it completely.

**Comment by tonnico** [comment]
#2103
